src/makeCalculator.js

```diff
@@ -4,7 +4,28 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add: (num) => calculator.result + num,
+    subtract: (num) => calculator.result - num,
+    multiply: (num) => calculator.result * num,
+    divide: (num) =>
+      num !== 0 ? calculator.result / num : 'Error: Division by zero',
+
+    operate(callback, num) {
+      calculator.result = callback(num);
+
+      return this;
+    },
+
+    reset() {
+      calculator.result = 0;
+
+      return this;
+    },
+  };
+
+  return calculator;
 }
```

Comment:
[NAMING] The function is called `makeCalculator`, so it would be better to return `calculator` at the end of the function. This will make your code more readable and maintainable.